adaptors task/student/include/impl.h

```diff
@@ -11,4 +13,64 @@
  *
  * @return true if expression is valid
  */
-bool isValid(const std::string& source);
+bool isOpenBracket(char ch, std::vector< std::pair <char,char> >& vector){
+    bool isOpen = false;
+    for(int j=0; j< vector.size();++j){
+        if(ch == vector[j].first)
+            isOpen = true;
+    }
+    return isOpen;
+}
+
+bool isClosedBracket(char ch, std::vector< std::pair <char,char> >& vector){
+    bool isOpen = false;
+    for(int j=0; j< vector.size();++j){
+        if(ch == vector[j].second)
+            isOpen = true;
+    }
+    return isOpen;
+}
+
+char correspondingClosedBracket(char ch, std::vector< std::pair <char,char> >& vector){
```

Comment:
I like your style, when you separate code to the corresponding functions, thanks!